homeassistant/components/microbees/binary_sensor.py

```diff
@@ -0,0 +1,82 @@
+"""BinarySensor integration microBees."""
+from microBeesPy import Sensor
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import MicroBeesUpdateCoordinator
+from .entity import MicroBeesEntity
+
+BINARYSENSOR_TYPES = {
+    12: BinarySensorEntityDescription(
+        device_class=BinarySensorDeviceClass.MOTION,
+        key="motion_sensor",
+    ),
+    13: BinarySensorEntityDescription(
+        device_class=BinarySensorDeviceClass.DOOR,
+        key="door_sensor",
+    ),
+    19: BinarySensorEntityDescription(
+        device_class=BinarySensorDeviceClass.MOISTURE,
+        key="moisture_sensor",
+    ),
+    20: BinarySensorEntityDescription(
+        device_class=BinarySensorDeviceClass.SMOKE,
+        key="smoke_sensor",
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Config entry."""
+    coordinator: MicroBeesUpdateCoordinator = hass.data[DOMAIN][
+        entry.entry_id
+    ].coordinator
+    async_add_entities(
+        MBBinarySensor(coordinator, entity_description, bee_id, binary_sensor.id)
+        for bee_id, bee in coordinator.data.bees.items()
+        for binary_sensor in bee.sensors
+        if (entity_description := BINARYSENSOR_TYPES.get(binary_sensor.device_type))
+        is not None
+    )
+
+
+class MBBinarySensor(MicroBeesEntity, BinarySensorEntity):
+    """Representation of a microBees BinarySensor."""
+
```

Comment:
has_entiy_name is inside MicroBeesEntity alrady

Reply:
Make sure you are looking at newer integrations, the scheme has changed.

In the case you are describing (@bdraco correct me if I'm wrong), you would create a light entity.  It would have an entity name of None. It would also have a device property filled in according to this:
[device properties](https://developers.home-assistant.io/docs/device_registry_index#device-properties)

Your identifier would probably be the serial number.   It's name would be the name and name and name_by_user properties.

you would add the switch status sensor binary sensor entity.  It would have a name like "Switch Status" as a name, and it would return the device object you created when you created the light entity as the device property.

Reply:
@rlippmann we haven't seen any Integration act like this. can you please suggest some sample? it's a new practice but I think is still valid the old one we have followed. Is this PR legit and ready to be merged?

Reply:
1) I have no authority to approve or deny a pull request.  I am merely giving you suggestions on how you would go about getting it approve
2) both those instances cited have self._attr_device_info attributes.

Reply:
@rlippmann we have too. please check MicroBeesEntity.